ContactMechanics/FFTElasticHalfSpace.py

```diff
@@ -755,23 +768,31 @@ def evaluate(self, disp, pot=True, forces=False):
             if pot:
                 potential = self.evaluate_elastic_energy(force, disp)
         elif pot:
-            kforce = self.evaluate_k_force(disp)
+            # kforce = self.evaluate_k_force(disp)
             # TODO: OPTIMISATION: here kdisp is computed twice, because it's
             #  needed in kforce
             self.real_buffer.array()[...] = disp
             self.fftengine.fft(self.real_buffer, self.fourier_buffer)
+            dispk = self.fourier_buffer.array()[...].copy()
```

Comment:
do you need this copy ? But I agree it is safer

Reply:
yes.. since I use this fourier buffer array multiple times in the same script. So to be on safe side.

Reply:
Ok. But you don't reuse the fourier buffer within this function. So I think it is safe not to copy. 
What I suggest is to make all tests pass, and then remove the copy and see if still all tests pass.

Reply:
okay! I will try doing that.